src/ucm/event/event.c

```diff
@@ -171,6 +171,8 @@ void ucm_event_leave()
     pthread_rwlock_unlock(&ucm_event_lock);
 }
 
+void *ucm_mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset) UCS_F_NOINLINE;
```

Comment:
why need to define a separate prototype?
can't we just add UCS_F_NOINLINE to existing functions definitions?

Reply:
That might be fine too, just I've seen more confusing syntax variations used online when people put attributes in function definitions, and I'm not confident what syntax is correct

Reply:
pls see example:
https://github.com/openucx/ucx/blob/e5ecf5c6e223c62bd99475adadb4cb0068d623fd/src/ucp/core/ucp_worker.c#L2728

Reply:
Okay, I'll change to that syntax.  The GNU attribute syntax documentation isn't super clear, but I did find a couple examples in their own examples section that use the above syntax so I guess it's okay.